nimble/host/audio/include/host/audio/ble_audio.h

```diff
@@ -0,0 +1,413 @@
+    /** Broadcast Stream Encryption */
+    BLE_AUDIO_PUBLIC_BCST_ANNOUNCEMENT_FEAT_ENCRYPTION = 1 << 0,
+
+    /** Standard Quality Public Broadcast Audio */
+    BLE_AUDIO_PUBLIC_BCST_ANNOUNCEMENT_FEAT_AUDIO_SQ = 1 << 1,
+
+    /** High Quality Public Broadcast Audio */
+    BLE_AUDIO_PUBLIC_BCST_ANNOUNCEMENT_FEAT_AUDIO_HQ = 1 << 2,
+};
+
+/** @brief Public Broadcast Announcement structure */
+struct ble_audio_public_bcst_announcement {
+    /** Public Broadcast Announcement features bitfield */
+    enum ble_audio_public_bcst_announcement_feat feat;
+
+    /** Metadata Length */
+    uint8_t metadata_len;
+
+    /** Metadata */
+    const uint8_t *metadata;
+};
+
+struct ble_audio_bcst_name {
+    /** Broadcast Name length */
+    uint8_t name_len;
+
+    /** Broadcast Name */
+    const char *name;
+};
+
+/**
+ * @defgroup ble_audio_events Bluetooth Low Energy Audio Events
+ * @{
+ */
+
+/** BLE Audio event: Broadcast Sink created */
+#define BLE_AUDIO_EVENT_BSNK_CREATED                    0
+
+/** BLE Audio event: Broadcast Sink destroyed */
+#define BLE_AUDIO_EVENT_BSNK_DESTROYED                  1
+
+/** BLE Audio event: Broadcast Sink PA Sync state changed */
+#define BLE_AUDIO_EVENT_BSNK_PA_SYNC_STATE_CHANGED      2
+
+/** BLE Audio event: Broadcast Sink Encryption state changed */
+#define BLE_AUDIO_EVENT_BSNK_ENC_STATE_CHANGED          3
+
+/** BLE Audio event: Broadcast Sink Sync state changed */
+#define BLE_AUDIO_EVENT_BSNK_SYNC_STATE_CHANGED         4
+
+/** BLE Audio event: Basic Audio Announcement */
+#define BLE_AUDIO_EVENT_BASIC_AUDIO_ANNOUNCEMENT        5
+
+/** BLE Audio event: Broadcast Announcement */
+#define BLE_AUDIO_EVENT_BCST_ANNOUNCEMENT               6
+/** @} */
+
+/** @brief Audio Broadcast Sink created event */
+struct ble_audio_event_bsnk_created {
+    /** Audio Broadcast Sink instance */
+    struct ble_audio_bsnk *snk;
+};
+
+/** @brief Audio Broadcast Sink destroyed event */
+struct ble_audio_event_bsnk_destroyed {
+    /** Audio Broadcast Sink instance */
+    struct ble_audio_bsnk *snk;
+};
+
+/** @brief Audio Broadcast Sink PA Sync state changed event */
+struct ble_audio_event_bsnk_pa_sync_state_changed {
+    /** Audio Broadcast Sink instance */
+    struct ble_audio_bsnk *snk;
+
+    /** Periodic Advertisement Sync state */
+    enum ble_audio_bsnk_pa_sync_state state;
+};
+
+/** @brief Audio Broadcast Sink Encryption state changed event */
+struct ble_audio_event_bsnk_enc_state_changed {
+    /** Audio Broadcast Sink instance */
+    struct ble_audio_bsnk *snk;
+
+    /** BIG Encryption state */
+    enum ble_audio_bsnk_enc_state state;
+
+    // Add Invalid (bad) code
+};
+
+/** @brief Audio Broadcast Sink Sync state changed event */
+struct ble_audio_event_bsnk_sync_state_changed {
+    /** Audio Broadcast Sink instance */
+    struct ble_audio_bsnk *snk;
+
+    /** Audio Broadcast Sync state */
+    enum ble_audio_bsnk_sync_state state;
+
+    /** BIS Synchronization */
+    uint32_t bis_sync;
```

Comment:
I think this should be an array - probably pointer with count of entries (subgroups)

Reply:
done